src/PPrintEngine.ml

```diff
@@ -58,39 +58,90 @@ class type output = object
 
 end
 
+(* ------------------------------------------------------------------------- *)
+
+(* Printing blank space (indentation characters). *)
+
+let blank_length =
+  80
+
+let blank_buffer =
+  String.make blank_length ' '
+
+let rec blanks output_fn n =
+  if n <= 0 then
+    ()
+  else if n <= blank_length then
+    output_fn blank_buffer 0 n
+  else begin
+    output_fn blank_buffer 0 blank_length;
+    blanks output_fn (n - blank_length)
+  end
+
+(* ------------------------------------------------------------------------- *)
+
+class with_blank_buffering char substring = object(self)
+  val mutable blank_buffered = None 
```

Comment:
The distinction between `None` and `Some 0` seems superfluous. Perhaps `blank_buffered` could have type `int` instead of `int option`?